### PR TITLE
Quic Transport with more thread safety

### PR DESCRIPTION
**Pull request by TimEvens**

### CODE REVIEW

File: src/quicr_client.cpp [hunk 1]
```diff
@@ -81,40 +97,45 @@ class QuicRTransportDelegate : public ITransport::TransportDelegate
...
+    const qtransport::StreamId& /* mStreamId */)
   {
   }
 
   virtual void on_recv_notify(const qtransport::TransportContextId& context_id,
-                              const qtransport::MediaStreamId& mStreamId)
+                              const qtransport::StreamId& streamId)
   {
-    // TODO: Consider running the below async or in a thread
+    for (int i=0; i < 60; i++) {
```
Comment (GhostofCookie) [inline src/quicr_client.cpp:109]: Noob question: Why 60?

File: src/quicr_client.cpp [hunk 2]
```diff
@@ -124,6 +145,7 @@ class QuicRTransportDelegate : public ITransport::TransportDelegate
 
 void
 QuicRClient::make_transport(RelayInfo& relay_info,
+                            qtransport::TransportConfig tconfig,
```
Comment (GhostofCookie) [inline src/quicr_client.cpp:148]: Should TransportConfig be marked as &&?

File: src/quicr_server.cpp [hunk 3]
```diff
@@ -69,10 +70,13 @@ QuicRServer::QuicRServer(std::shared_ptr<qtransport::ITransport> transport_in,
 }
 
 std::shared_ptr<qtransport::ITransport>
-QuicRServer::setupTransport([[maybe_unused]] RelayInfo& relayInfo)
+QuicRServer::setupTransport([[maybe_unused]] RelayInfo& relayInfo,
+                            qtransport::TransportConfig cfg)
```
Comment (GhostofCookie) [inline src/quicr_server.cpp:74]: Should this be &&? Same question for line 39 as well.

File: src/quicr_server.cpp [hunk 4]
```diff
@@ -264,6 +280,37 @@ QuicRServer::TransportDelegate::on_connection_status(
   log_msg << "connection_status: cid: " << context_id
           << " status: " << int(status);
   server.log_handler.log(qtransport::LogLevel::debug, log_msg.str());
+
+  if (status == qtransport::TransportStatus::Disconnected) {
+    log_msg.str("");
```
Comment (GhostofCookie) [inline src/quicr_server.cpp:393]: Seems odd, is this just a missed deletion?

**Review by GhostofCookie — DISMISSED** [review]
Looks good to me.

File: src/quicr_client.cpp (hunk 1 above)
Comment (TimEvens) [reply src/quicr_client.cpp:109]: Just a starting point. We don’t want to block the event loop thread, so this does a little at a time. We’ll need to refactor this based on either config or breaking this out into threads or something else.

File: src/quicr_client.cpp (hunk 2 above)
Comment (TimEvens) [reply src/quicr_client.cpp:148]: In terms of performance, this is not an issue. A copy should be fine. We cannot use ```std::move()``` considering the caller can/will reuse the config for multiple instances of the client/server. Laps is an example of this where it uses the same config and passes it with initial settings to the first server instance and then minor changes to the second (e.g., port and TLS cert changes).

File: src/quicr_server.cpp (hunk 3 above)
Comment (TimEvens) [reply src/quicr_server.cpp:74]: Same comment above.

File: src/quicr_server.cpp (hunk 4 above)
Comment (TimEvens) [reply src/quicr_server.cpp:393]: ```log_msg.str(“”)``` reinitializes the output string stream. Might be more efficient with ```string()``` but in terms of performance, this is not a concern.

**Review by GhostofCookie — APPROVED** [review]

**Pull request merged**